File: ios/EnterpriseMobileApp/Sources/Core/Domain/Repositories/RewardsRepository.swift
import Foundation

/// Repository protocol for rewards operations.
public protocol RewardsRepository: Sendable {
    func getRewards() -> AsyncStream<DomainResult<[Reward]>>
    func getAccount() async -> DomainResult<RewardsAccount>
    func redeemReward(rewardId: String) async -> DomainResult<RewardsAccount>
}

File: android/core/domain/src/main/java/com/enterprise/core/domain/repositories/RewardsRepository.kt
package com.enterprise.core.domain.repositories

import com.enterprise.core.domain.entities.DomainResult
import com.enterprise.core.domain.entities.Reward
import com.enterprise.core.domain.entities.RewardsAccount
import kotlinx.coroutines.flow.Flow

/** Rewards program repository contract. */
interface RewardsRepository {
    fun getRewards(): Flow<DomainResult<List<Reward>>>
    suspend fun getAccount(): DomainResult<RewardsAccount>
    suspend fun redeemReward(rewardId: String): DomainResult<RewardsAccount>
}
